examples/animation/animation_graph.rs

```diff
@@ -0,0 +1,538 @@
+//! Demonstrates animation blending with animation graphs.
+//!
+//! The animation graph is shown on screen. You can change the weights of the
+//! playing animations by clicking and dragging left or right within the nodes.
+
+use bevy::{
+    color::palettes::{
+        basic::WHITE,
+        css::{ANTIQUE_WHITE, DARK_GREEN},
+    },
+    prelude::{Color::Srgba, *},
+};
+
+/// An on-screen representation of a node.
+#[derive(Debug)]
+struct NodeRect {
+    /// The number of pixels that this rectangle is from the left edge of the
+    /// window.
+    left: f32,
+    /// The number of pixels that this rectangle is from the bottom edge of the
+    /// window.
+    bottom: f32,
+    /// The width of this rectangle in pixels.
+    width: f32,
+    /// The height of this rectangle in pixels.
+    height: f32,
+}
+
+/// Either a straight horizontal or a straight vertical line on screen.
+///
+/// The line starts at (`left`, `bottom`) and goes either right (if the line is
+/// horizontal) or down (if the line is vertical).
+struct Line {
+    /// The number of pixels that the start of this line is from the left edge
+    /// of the screen.
+    left: f32,
+    /// The number of pixels that the start of this line is from the bottom edge
+    /// of the screen.
+    bottom: f32,
+    /// The length of the line.
+    length: f32,
+}
+
+/// Assets that the example needs to use.
+#[derive(Resource)]
+struct AppAssets {
+    /// The [`AnimationGraph`] asset, which specifies how the animations are to
+    /// be blended together.
+    animation_graph: Handle<AnimationGraph>,
+    /// Indices of the three animations in the graph.
+    ///
+    /// This value is populated when the graph is loaded.
+    node_indices: [AnimationNodeIndex; 3],
+}
+
+/// The current weights of the three playing animations.
```

Comment:
Doesn't this make more sense as a component on the root entity?

Reply:
Done